ocaml-lsp-server/src/custom_requests/req_hover_extended.ml

```diff
@@ -0,0 +1,70 @@
+open Import
+open Fiber.O
+
+let capability = ("handleHoverExtended", `Bool true)
+
+let meth = "ocamllsp/hoverExtended"
+
+module Request_params = struct
+  type t =
+    { text_document : TextDocumentIdentifier.t
+    ; cursor_position : Position.t
+    ; verbosity : int Json.Nullable_option.t
+          [@default None] [@yojson_drop_default ( = )]
+    }
+  [@@deriving_inline yojson] [@@yojson.allow_extra_fields]
```

Comment:
I wrote of_jsonrpc_params by hand as it was done in another custom req, but is there a reason not to use deriving instead?

Reply:
You can use either. The workflow with promotion is quite annoying though.